--- lpSolve/lpSolve/src/commonlib.c.orig	2012-02-20 05:34:23.059387355 -0200
+++ lpSolve/lpSolve/src/commonlib.c	2012-02-20 05:36:22.371008611 -0200
@@ -688,7 +688,7 @@ void blockWriteINT(FILE *output, char *l
 {
   int i, k = 0;
 
-  fprintf(output, label);
+  fprintf(output, "%s", label);
   fprintf(output, "\n");
   for(i = first; i <= last; i++) {
     fprintf(output, " %5d", myvector[i]);
@@ -707,7 +707,7 @@ void blockWriteBOOL(FILE *output, char *
 {
   int i, k = 0;
 
-  fprintf(output, label);
+  fprintf(output, "%s", label);
   fprintf(output, "\n");
   for(i = first; i <= last; i++) {
     if(asRaw)
@@ -729,7 +729,7 @@ void blockWriteREAL(FILE *output, char *
 {
   int i, k = 0;
 
-  fprintf(output, label);
+  fprintf(output, "%s", label);
   fprintf(output, "\n");
   for(i = first; i <= last; i++) {
     fprintf(output, " %18g", myvector[i]);
--- lpSolve/lpSolve/src/hbio.c.orig	2012-02-20 05:36:53.284169218 -0200
+++ lpSolve/lpSolve/src/hbio.c	2012-02-20 05:37:21.483315544 -0200
@@ -1602,7 +1602,7 @@ void upcase(char* S)
 
 void IOHBTerminate(char* message)
 {
-   fprintf(stderr,message);
+   fprintf(stderr,"%s",message);
    exit(1);
 }
 
--- lpSolve/lpSolve/src/lp_report.c.orig	2012-02-20 05:37:58.348506666 -0200
+++ lpSolve/lpSolve/src/lp_report.c	2012-02-20 05:38:34.963696256 -0200
@@ -160,7 +160,7 @@ void blockWriteLREAL(FILE *output, char
 {
   int i, k = 0;
 
-  fprintf(output, label);
+  fprintf(output, "%s", label);
   fprintf(output, "\n");
   for(i = first; i <= last; i++) {
     fprintf(output, " %18g", vector[i]);
@@ -189,7 +189,7 @@ void blockWriteAMAT(FILE *output, const
   if(last < 0)
     last = lp->rows;
 
-  fprintf(output, label);
+  fprintf(output, "%s", label);
   fprintf(output, "\n");
 
   if(first == 0) {
@@ -254,7 +254,7 @@ void blockWriteBMAT(FILE *output, const
   if(last < 0)
     last = lp->rows;
 
-  fprintf(output, label);
+  fprintf(output, "%s", label);
   fprintf(output, "\n");
 
   for(i = first; i <= last; i++) {
